php-java/advanced-operations/working-with-pages/stamping/text-stamps-in-pdf-file/_index.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Add Text stamps in PDF programmatically
+linktitle: Text stamps in PDF File
+type: docs
+weight: 20
+url: /php-java/text-stamps-in-the-pdf-file/
+description: Add a text stamp to a PDF file using the TextStamp class with PHP.
+lastmod: "2024-06-05"
+sitemap:
+    changefreq: "weekly"
+    priority: 0.7
+---
+
+## Add Text Stamp with Java
+
+Aspose.PDF for PHP via Java provides [TextStamp](https://reference.aspose.com/pdf/java/com.aspose.pdf/TextStamp) class to add a text stamp in a PDF file. The [TextStamp](https://reference.aspose.com/pdf/java/com.aspose.pdf/TextStamp) class provides necessary methods to specify font size, font style, and font color etc for stamp object. In order to add text stamp, first you need to create a [Document](https://reference.aspose.com/pdf/java/com.aspose.pdf/Document) object and a [TextStamp](https://reference.aspose.com/pdf/java/com.aspose.pdf/TextStamp) object using required methods. After that, you may call [addStamp(..)](https://reference.aspose.com/pdf/java/com.aspose.pdf/Page#addStamp-com.aspose.pdf.Stamp-) method of the [Page](https://reference.aspose.com/pdf/java/com.aspose.pdf/Page) class to add the stamp in the PDF document.
+
+The following code snippet shows you how to add text stamp in the PDF file.
+
+```php
+
+    // Open document
+    $document = new Document($inputFile);        
+    $pages = $document->getPages();
+    $colors = new Color();
+    // create text stamp
+    $textStamp = new TextStamp("Sample Stamp");
+    // set whether stamp is background
+    $textStamp->setBackground(true);
+    // set origin
+    $textStamp->setXIndent(100);
+    $textStamp->setYIndent(100);
+    // rotate stamp
+    $textStamp->setRotate((new Rotation())->getOn90());
+    
+    // set text properties
+    $fontRepository = new FontRepository();
+    $textStamp->getTextState()->setFont($fontRepository->findFont("Arial"));
+    $textStamp->getTextState()->setFontSize(14);
+    $textStamp->getTextState()->setFontStyle(FontStyles::$Bold | FontStyles::$Italic);
+    $textStamp->getTextState()->setForegroundColor($colors->getGreen());
+    
+    // add stamp to particular page
+    $pages->get_Item(1)->addStamp($textStamp);
+
+    // Save output document
+    $document->save($outputFile);
+    $document->close();
```

Comment:
// Open document
    $document = new Document($inputFile);        
    $pages = $document->getPages();
    $colors = new Color();
    // create text stamp
    $textStamp = new TextStamp("Sample Stamp");
    // set whether stamp is background
    $textStamp->setBackground(true);
    // set origin
    $textStamp->setXIndent(100);
    $textStamp->setYIndent(100);
    // rotate stamp
    $textStamp->setRotate((new Rotation())->On90);    
    // set text properties
    $fontRepository = new FontRepository();
    $fontStyles = new FontStyles();
    $textStamp->getTextState()->setFont($fontRepository->findFont("Arial"));
    $textStamp->getTextState()->setFontSize(14);
    $textStamp->getTextState()->setFontStyle($fontStyles->Bold | $fontStyles->Italic);
    $textStamp->getTextState()->setForegroundColor($colors->getGreen());
    
    // add stamp to particular page
    $pages->get_Item(1)->addStamp($textStamp);

    // Save output document
    $document->save($outputFile);
    $document->close();